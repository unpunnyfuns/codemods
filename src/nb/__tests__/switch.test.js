import { describe, expect, it } from 'vitest'
import { DEFAULT_TEST_OPTIONS, testTransform } from '../../__tests__/test-helper.js'

describe('nb/switch', () => {
  it('migrates Switch to Nordlys with children wrapping and prop transformations', () => {
    const output = testTransform('nb/switch', 'nb/switch', 'tsx', {
      sourceImport: DEFAULT_TEST_OPTIONS.sourceImport,
      targetImport: '@target/components/Switch',
      tokenImport: DEFAULT_TEST_OPTIONS.tokenImport,
    })
    expect(output).toMatchSnapshot()
  })
})
