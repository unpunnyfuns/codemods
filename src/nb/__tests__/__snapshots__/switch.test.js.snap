// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nb/switch > migrates Switch to Nordlys with children wrapping and prop transformations 1`] = `
"import { Typography } from '@source/components';

import { Switch } from '@target/components/Switch';

export function MyComponent() {
  return (
    <>

      {/* Dropped during migration (not supported in React Native):
        switchPosition="right"
        hStackProps={...}
        label="Additional info" */
      }
      <Switch testID="switch1" value={true} onValueChange={(val) => console.log(val)}>
        <Switch.Label>
          <Typography type="heading" size="sm">
            Main label
          </Typography>
        </Switch.Label>
        <Switch.Description>Additional info</Switch.Description>
      </Switch>
      <Switch value={false} onValueChange={() => {}} disabled>
        <Switch.Label>Simple switch
                </Switch.Label>
      </Switch>
    </>
  );
}
"
`;
